src/tools/rbd_wnbd/rbd_wnbd.cc

```diff
@@ -1159,8 +1195,24 @@ static int do_map(Config *cfg)
     global_init_postfork_finish(g_ceph_context);
   }
 
-  handler->wait();
-  handler->shutdown();
+  {
+    uint64_t watch_handle;
+    WNBDWatchCtx watch_ctx = WNBDWatchCtx(io_ctx, handler, image, info.size);
+    r = image.update_watch(&watch_ctx, &watch_handle);
+    if (r < 0)
+      derr << __func__ << ": update_watch failed with error: "
+          << cpp_strerror(r) << dendl;
```

Comment:
So instead of proper cleanup, the change is to just swallow `update_watch` errors?  This means that this mapping wouldn't get any resize notifications.  @petrutlucian94 are you OK with that?

Reply:
We should definitely clean up and error out in case of ``update_watch`` errors. This was probably an oversight.

Reply:
> Wouldn't this jump straight to closing image and tearing down rados under a running handler and also potentially leave a stale registry entry behind?

Even before this PR, some registry entries were left in place intentionally in case of wnbd failures so that "rbd device list" would show an "inactive" disk [1]:

```
PS C:\Users\Administrator> rbd device ls
id  pool  namespace  image       snap  device          disk_number  status
-1  rbd              rbd_win_1g  -     rbd/rbd_win_1g  -1           inactive
```

On a second thought, it probably makes more sense to always remove the registry entry in case of failures when dealing with non-persistent mappings. We can just move that "close_ret" label above the registry cleanup.

Mappings that are marked as "persistent" will never be removed from the registry unless explicitly unmapped, even in case of failures. The "ceph-rbd" service will try to reconnect them (e.g. when the server boots) and we wouldn't want those mappings to be unregistered if that fails.

[1] https://github.com/ceph/ceph/blob/main/src/tools/rbd_wnbd/rbd_wnbd.cc#L1125-L1136